### PR TITLE
feat(integrations): add support for fal.ai

### PR DESCRIPTION
**Pull request by Shubhamai**
## Describe your changes
- Added fal.ai support.

## Issue ticket number and link
https://github.com/NangoHQ/nango/issues/2767

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -5151,3 +5151,18 @@ zoominfo:
     token_request_auth_method: custom
     body_format: json
     docs: https://docs.nango.dev/integrations/all/zoominfo
+
+
+falai:
+    display_name: fal.ai
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:5265]: missing categories (I guess we should enforce it in the CI)

File: packages/shared/providers.yaml [hunk 2]
```diff
@@ -5151,3 +5151,18 @@ zoominfo:
...
+    auth_mode: API_KEY
+    proxy:
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:5270]: can you add a `verification` prop please ☺️

**Review by bodinsamuel — COMMENTED** [review]
🚀

**Comment by hassan254-prog** [comment]
same as here, please use this [logo](https://codefile.io/f/yYgBgU6fHp)

**Comment by Shubhamai** [comment]
Thanks, made the review changes. Except review prop. I don't think there's endpoint in fal.ai to verify the API Key without causing unintended effects.

**Comment by bodinsamuel** [comment]
>Thanks, made the review changes. Except review prop. I don't think there's endpoint in fal.ai to verify the API Key without causing unintended effects.

Mmh indeed that's unfortunate

**Review by bodinsamuel — APPROVED** [review]

**Review by hassan254-prog — APPROVED** [review]

**Pull request merged**